dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/internal/JwtCache.java

```diff
@@ -23,33 +22,37 @@
 import com.ibm.websphere.ras.TraceComponent;
 import com.ibm.websphere.ras.annotation.Sensitive;
 import com.ibm.ws.ffdc.annotation.FFDCIgnore;
-import com.ibm.ws.security.common.structures.CacheEntry;
 import com.ibm.ws.security.common.structures.SingleTableCache;
-import com.ibm.ws.security.jwt.config.JwtConsumerConfig;
 
 public class JwtCache extends SingleTableCache {
 
     private static final TraceComponent tc = Tr.register(JwtCache.class);
 
     static final int DEFAULT_ENTRY_LIMIT = 500;
 
-    private final JwtConsumerConfig config;
-
-    public JwtCache(long timeoutInMilliSeconds, JwtConsumerConfig config) {
+    public JwtCache(long timeoutInMilliSeconds) {
         super(DEFAULT_ENTRY_LIMIT, timeoutInMilliSeconds);
-        this.config = config;
     }
 
     /**
      * Find and return the object associated with the specified key.
      */
-    @Override
-    public synchronized Object get(@Sensitive String key) {
-        JwtContext jwtContext = (JwtContext) super.get(key);
-        if (jwtContext == null || isJwtExpired(jwtContext)) {
+    public synchronized Object get(@Sensitive String jwt, String configId) {
+        JwtCacheKey key = getCacheKey(jwt, configId);
+        JwtCacheValue cacheValue = (JwtCacheValue) get(key);
+        if (cacheValue == null || isJwtExpired(cacheValue)) {
             return null;
         }
-        return jwtContext;
+        return cacheValue.getValue();
+    }
+
+    /**
+     * Insert the value into the Cache using the specified key.
+     */
+    public synchronized void put(@Sensitive String jwt, String configId, JwtContext value, long clockSkew) {
+        JwtCacheKey key = getCacheKey(jwt, configId);
+        JwtCacheValue cacheValue = new JwtCacheValue(value, clockSkew);
+        super.put(key, cacheValue, clockSkew);
```

Comment:
Similarly I suggest:

```suggestion
    public void put(@Sensitive String jwt, String configId, JwtContext value, long clockSkew) {
        JwtCacheKey key = getCacheKey(jwt, configId);
        super.put(key, value, clockSkew);
    }

    @Override
    protected CacheValue createCacheValue(Object jwtContext, long clockSkew) {
        return JwtCacheValue(jwtContext, clockSkew);
    }
```

Note the removal of synchronized again.